### PR TITLE
Split out validation logic for categorizer

### PR DESCRIPTION
**Pull request by Myranae**
## Summary:
For the servier-side scoring project, we are moving validation logic out of each widget component. This moves the validation logic for the categorizer widget.

Issue: LEMS-2341

## Test plan:
- All tests should pass
- Nothing should chang as code is just being moved into separate files

### CODE REVIEW

File: packages/perseus/src/widgets/categorizer/categorizer-validator.test.ts [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+                "Nonlinear relationship",
+            ],
+            highlightLint: false,
+            static: false,
+        };
+
+        const useInput = {
+            values: [1, 3],
+        } as const;
+        const score = Categorizer.validate(useInput, rubric, mockStrings);
```
Comment (handeyeco) [inline packages/perseus/src/widgets/categorizer/categorizer-validator.test.ts:116]: We need to replace all instances of `[Widget].validate` and `[Widget].simpleValidate` with the pure `widgetValidator` function. No need to go looking for uses, but in files we touch we should be making this change.

**Review by handeyeco — CHANGES_REQUESTED** [review]
Same feedback as from input-number, need to completely decouple from React and ideally replace object snapshot tests with `.toEqual`.

Also replace component method calls with the pure validator where possible.

File: packages/perseus/src/widgets/categorizer/categorizer-validator.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import type {Rubric, UserInput} from "./categorizer.types";
+import type {PerseusStrings} from "../../strings";
+import type {PerseusScore} from "@khanacademy/perseus";
```
Comment (handeyeco) [inline packages/perseus/src/widgets/categorizer/categorizer-validator.ts:3]: We should avoid importing from `@khanacademy/perseus` when we're within `@khanacademy/perseus`. The solution is relative imports.

**Review by handeyeco — APPROVED** [review]
Just a small ask regarding the import

File: packages/perseus/src/widgets/categorizer/categorizer-validator.ts (hunk 2 above)
Comment (Myranae) [reply packages/perseus/src/widgets/categorizer/categorizer-validator.ts]: 🤦‍♀️ Thank you! Fixed!

**Pull request merged**